### ISSUE TITLE
[Android] `TextBox.KeyDown` event is not working

### ISSUE THREAD

**Issue by Soap-141** [unoplatform/uno#13983]
### Current behavior

The `KeyDown` event is not triggered on my Google Pixel 6. It may be working on Samsung devices.

### Expected behavior

The `KeyDown` event  should triggered on all devices.

### How to reproduce it (as minimally and precisely as possible)

1. Download [TextBoxEvent.zip](https://github.com/unoplatform/uno/files/12918787/TextBoxEvent.zip)
2. Add a breakpoint in the event handler
3. Start project on Android
4. Press key on virtual keyboard (Enter and others)
5. Notice the event is not triggered

### Workaround

N/A

### Works on UWP/WinUI

None

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia, Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

Latest stable 4.10 because I haven't been able to build latest preview.

### Affected platforms

Android

### IDE

Visual Studio 2022

### IDE version

Latest

### Relevant plugins

N/A

### Anything else we need to know?

N/A

**Comment by jeromelaban** [comment]
Thanks for the report. It is likely that the KeyDown event is not propagated (and may not be able to) because the TextBox is handling it for itself.

Is it something that worked before?

**Comment by Soap-141** [comment]
@jeromelaban Yes it did, I can confirm it soon, I'll go back to a previous version of the project and tell you which Uno version it was. But if it's related to the device itself and might not work in previous versions. I'll update you in 15 minutes.

**Comment by Youssef1313** [comment]
https://developer.android.com/reference/android/view/KeyEvent.html

> As soft input methods can use multiple and inventive ways of inputting text, there is no guarantee that any key press on a soft keyboard will generate a key event: this is left to the IME's discretion, and in fact sending such events is discouraged. You should never rely on receiving KeyEvents for any key on a soft input method. In particular, the default software keyboard will never send any key event to any application targetting Jelly Bean or later, and will only send events for some presses of the delete and return keys to applications targetting Ice Cream Sandwich or earlier. Be aware that other software input methods may never send key events regardless of the version. Consider using editor actions like [EditorInfo.IME_ACTION_DONE](https://developer.android.com/reference/android/view/inputmethod/EditorInfo#IME_ACTION_DONE) if you need specific interaction with the software keyboard, as it gives more visibility to the user as to how your application will react to key presses

https://developer.android.com/develop/ui/views/touch-and-input/keyboard-input/commands

> Note: When handling keyboard events with the [KeyEvent](https://developer.android.com/reference/android/view/KeyEvent) class and related APIs, expect that the keyboard events are coming only from a hardware keyboard. Never rely on receiving key events for any key on a soft input method (an on-screen keyboard).

There doesn't seem to be a way for us to reliably detect input from soft keyboard. This feels like a platform limitation that we can't do anything about.

**Comment by MartinZikmund** [comment]
# Repro generator

Runtime test(s) added in branch [`dev/mazi/repro/13983`](https://github.com/unoplatform/uno/compare/dev/mazi/repro/13983):
- `src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBox.cs` → `When_TextBox_Focused_And_Key_Pressed_Then_KeyDown_Fires`
- `src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBox.cs` → `When_TextBox_Focused_And_Enter_Pressed_Then_KeyDown_Fires`

⚠️ The test(s) **pass on Skia targets**. The issue reports the bug on **Android** (Google Pixel 6), which is a native target not covered by the Skia runtime test runner. This bug is likely **native-target-specific** and may not reproduce on Skia. A PR has been created to add the passing test(s) as regression coverage.

Note: The tests use `OnKeyTestingOnly` which bypasses the virtual keyboard (IME) layer. The actual Android bug may be in the IME event routing path, which requires device-level testing to verify.
